api/v1beta2/artifact_types.go

```diff
@@ -46,6 +49,11 @@ type Artifact struct {
 	// +optional
 	Checksum string `json:"checksum"`
 
+	// ExcludedPatternsChecksum is a SHA256 checksum of the excluded patterns
+	// applied in the last reconciliation.
+	// +optional
+	ExcludedPatternsChecksum string `json:"excludedPatternsChecksum,omitempty"`
```

Comment:
Not every object is making use of exclusions. Given this, I do not think this field should be added to the Artifact, but rather to the Status objects from the Bucket and GitRepository.

In addition, I do not think that the new term `ExcludedPatterns` should be introduced. I would rather call it `IgnoreChecksum` or `ObservedIgnore`.